packages/api/src/utils/authenticateUser.ts

```diff
@@ -0,0 +1,25 @@
+import { Response , Request } from 'express';
+
+type OAuthUserData = {
+    email: string ,
+    firstName?: string ,
+    lastName?: string ,
+    phoneNumber?: string ,
+    address?: string ,
+    given_name?: string ,
+    family_name?: string ,
+}
+
+const authenticateUser = ({ options , request:req , response:res }:{options:OAuthUserData,request:Request , response:Response}) => {
+    // validate the request has a valid OAuthUserData object
+    // if so, add it to the session
+    // if not, redirect to error
+    if (options.email) {
+        req.session = options
+        res.redirect('/')
+    } else {
+        res.redirect('/error')
+    }
+}
+
+export { authenticateUser , OAuthUserData }
```

Comment:
While this syntax is totally valid, I highly prefer `export` to be used where the items are being declared as it increases transparency around what is and what is not exported. If you feel strongly about this syntax, let's discuss as a team but I want to standardize one way or the other.
```suggestion
```